signer/src/transaction_coordinator.rs

```diff
@@ -390,6 +392,75 @@ where
         .map(|_| ())
     }
 
+    /// Constructs a BitcoinPreSignRequest from the given transaction package and
+    /// sends it to the signers. Waits for acknowledgments from the signers until
+    /// the threshold is met or a timeout occurs.
+    /// If the signal stream closes unexpectedly, triggers a shutdown.
+    #[tracing::instrument(skip_all)]
+    async fn construct_and_send_bitcoin_presign_request(
+        &mut self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+        pending_requests: &SbtcRequests,
+        transaction_package: &[utxo::UnsignedTransaction<'_>],
+    ) -> Result<(), Error> {
+        // Create the BitcoinPreSignRequest from the transaction package
+        let sbtc_requests = BitcoinPreSignRequest {
+            request_package: transaction_package
+                .iter()
+                .map(|tx| (&tx.requests).into())
+                .collect(),
+            fee_rate: pending_requests.signer_state.fee_rate,
+            last_fees: pending_requests.signer_state.last_fees.map(Into::into),
+        };
+        let presign_ack_filter = {
+            let target_tip = *bitcoin_chain_tip;
+            move |event: &SignerSignal| {
+                matches!(
+                    event,
+                    SignerSignal::Event(SignerEvent::TxSigner(TxSignerEvent::MessageGenerated(
+                        Signed {
+                            inner: SignerMessage {
+                                bitcoin_chain_tip,
+                                payload: Payload::BitcoinPreSignAck(_),
+                            },
+                            ..
+                        },
+                    ))) if *bitcoin_chain_tip == target_tip
```

Comment:
I don't think that we should do the chain-tip filtering here, unless we also log that we've received a message where the chain tip is not what we are expecting. For example, in `drive_wsts_state_machine` we log that the coordinator observes "concurrent WSTS activity"

Reply:
The idea is to have an identified to be sure that the ack we received is connected to this request. We considered adding a unique identifier to the `BitcoinPreSignAck` to link it with the corresponding `BitcoinPreSignRequest`. But, we agreed that using the `bitcoin_chain_tip` as the key check is sufficient, since we are going to only send a single PreSignRequest per new block.

I’ll add a log for cases where the chain tip doesn’t match the expected value though. But if you feel that this doesn't make any sense I can get rid of that!

Reply:
Yeah, that's the right idea. I was suggesting that we are just as noisy here as we are in other places where we assume the same thing. So yeah, definitely do the filtering, but log when the bitcoin chain tip is not what we expect.

Reply:
implemented in https://github.com/stacks-network/sbtc/pull/956/commits/1818278009b40028f7518c3c446eb9dab7ec0118